src/main/java/org/openstreetmap/atlas/geography/atlas/change/FeatureChange.java

```diff
@@ -47,61 +47,82 @@
-    private static final BinaryOperator<Set<Long>> directReferenceMerger = Sets::withSets;
-    private static final BinaryOperator<SortedSet<Long>> directReferenceMergerSorted = Sets::withSortedSets;
-    private static final BinaryOperator<Set<Long>> directReferenceMergerLoose = (left,
-            right) -> Sets.withSets(false, left, right);
-    private static final BinaryOperator<RelationBean> relationBeanMerger = RelationBean::merge;
 
     private final ChangeType changeType;
-    private final AtlasEntity reference;
+    private final AtlasEntity updatedView;
+    private AtlasEntity beforeView;
 
-    public static FeatureChange add(final AtlasEntity reference)
+    public static FeatureChange add(final AtlasEntity updatedView)
     {
-        return new FeatureChange(ChangeType.ADD, reference);
+        return new FeatureChange(ChangeType.ADD, updatedView);
+    }
+
+    public static FeatureChange add(final AtlasEntity updatedView, final Atlas atlasContext)
+    {
+        return new FeatureChange(ChangeType.ADD, updatedView).withAtlasContext(atlasContext);
     }
 
     public static FeatureChange remove(final AtlasEntity reference)
     {
         return new FeatureChange(ChangeType.REMOVE, reference);
     }
 
-    public FeatureChange(final ChangeType changeType, final AtlasEntity reference)
+    /**
+     * This constructor is package private, and should be used for unit testing purposes only. Users
+     * who wish to specify a before view should be using
+     * {@link FeatureChange#withAtlasContext(Atlas)}.
+     *
+     * @param changeType
+     *            the change type
+     * @param updatedView
+     *            the updated entity
+     * @param beforeView
+     *            the before entity
+     */
+    FeatureChange(final ChangeType changeType, final AtlasEntity updatedView,
+            final AtlasEntity beforeView)
     {
-        if (reference == null)
+        if (updatedView == null)
         {
             throw new CoreException("reference cannot be null.");
         }
-        if (!(reference instanceof CompleteEntity))
+        if (!(updatedView instanceof CompleteEntity))
         {
             throw new CoreException(
                     "FeatureChange requires CompleteEntity, found reference of type {}",
-                    reference.getClass().getName());
+                    updatedView.getClass().getName());
         }
         if (changeType == null)
         {
             throw new CoreException("changeType cannot be null.");
         }
         this.changeType = changeType;
-        this.reference = reference;
+        this.updatedView = updatedView;
+        this.beforeView = beforeView;
         this.validateUsefulFeatureChange();
     }
 
+    public FeatureChange(final ChangeType changeType, final AtlasEntity updatedView)
+    {
+        this(changeType, updatedView, null);
+    }
+
     @Override
     public Rectangle bounds()
```

Comment:
This method will need to be updated. Before/after bounding box aggregation logic previously assigned to `CompleteEntities` will need to be performed here. This means will now need to strictly enforce `Located`ness for `CompleteEntities`. More on this below.